examples/http-rust-router/src/lib.rs

```diff
@@ -17,15 +18,17 @@ mod api {
     use super::*;
 
     // /hello/:planet
-    pub fn hello_planet(_req: Request, params: Params) -> Result<impl IntoResponse> {
+    pub async fn hello_planet(_req: Request, params: Params) -> Result<impl IntoResponse> {
         let planet = params.get("planet").expect("PLANET");
-
         Ok(Response::new(200, planet.to_string()))
     }
 
     // /*
-    pub fn echo_wildcard(_req: Request, params: Params) -> Result<impl IntoResponse> {
+    pub fn echo_wildcard(
+        _req: Request,
+        params: Params,
+    ) -> impl Future<Output = Result<impl IntoResponse>> {
```

Comment:
I would make this an `async` function instead of having to do `impl Future`

Reply:
Yep! Using the `async` keyword here defin8litely has better ergonomics.